pkg/scheduler/core/generic_scheduler.go

```diff
@@ -650,17 +650,23 @@ func PrioritizeNodes(
 			results[i] = make(schedulerapi.HostPriorityList, len(nodes))
 		}
 	}
+
 	processNode := func(index int) {
 		nodeInfo := nodeNameToInfo[nodes[index].Name]
 		var err error
 		for i := range priorityConfigs {
+			// when error occurs, initialize the rest result[i] with default node name
+			if err != nil {
```

Comment:
Why do we need this? With this block, if an error occurs for one iteration, the next iteration will be skipped.

Reply:
My reading is that "if err happens, ensure further priorityConfigs (starting i+1) has a chance to initialize" - that's why @DylanBLE removed the `return` in L663 in this closure.

Reply:
I am fine with the removal of the return, but this if statement does not look correct to me.

Reply:
I see @bsalamat 's point - even if priorityConfig[i] failed, priorityConfig[i+1] should have its chance to run entire logic, rather than simply initialize host name and continue.

Reply:
Yeah, I agree we don't this block unless there is a chance of `err` showing up from other places.

Reply:
I see. My intention was to skip the **unnecessary** calculation of `Map` when error happens because of the `return schedulerapi.HostPriorityList{}, errors.NewAggregate(errs) ` at line 688 which will abandon all results. Maybe we can open another **pr** to optimize it.

Reply:
Got it.